ports/rp2/machine_uart.c

```diff
@@ -230,6 +272,14 @@ STATIC mp_obj_t machine_uart_make_new(const mp_obj_type_t *type, size_t n_args,
         self->invert = args[ARG_invert].u_int;
     }
 
+    // Set hardware flow control if configured.
+    if (args[ARG_flow].u_int > 0) {
```

Comment:
won't this need to be `>= 0`, so the user can disable flow control?

Reply:
That's right it's fixed now. But now I realized there may be some issues with this driver. The way this is done a uart object is created twice the second time without args, it will Not re-initialize the uart which means it will use the stored/previous baudrate/bits/etc and not the default values. And if args are used, `irq_set_exclusive_handler` will be called twice which I think results in some assertion failing. I haven't confirmed any of this, but I can test it later though.

Reply:
Ok, if that is a problem then it's a separate issue.  This PR is good to go now.

Reply:
It's true, irq_set_exclusive_handler is called every time if args > 1:

```Python
>>> from machine import UART
>>> uart = UART(1, 9600) 
machine_uart.c: irq handler 0x0
>>> uart = UART(1, 9600)
machine_uart.c: irq handler 0x100269d5
```
But re-installing the exclusive handler does not seem to assert as documented here: https://raspberrypi.github.io/pico-sdk-doxygen/group__hardware__irq.html#gafffd448ba2d2eef5b355b88180aefe7f

Maybe assertions are disabled, should this fixed anyway ?

Reply:
It shouldn't assert because it's reinstalling the same handler, and the check in tho pico-sdk is:
```c
hard_assert(current == __unhandled_user_irq || current == handler);
```

Reply:
> Also if no args are used, the last config is used instead of the defaults (is this intended?)

Yes that's intended, so you can retrieve an existing object without changing its config.

Reply:
> It shouldn't assert because it's reinstalling the same handler, and the check in tho pico-sdk is:

It does not, but that's not exactly what's documented:

> This method will assert if there is already any sort of interrupt handler installed for the specified irq number.

Reply:
I'm not sure if the docs are outdated, or the code is wrong, but it does seem to be logically doing the right thing: only allowing one (an exclusive) handler to be set (even if it's set multiple times, it's still exclusive).

Reply:
Okay, sounds good for now.